src/tools/rbd/action/DiskUsage.cc

```diff
@@ -217,19 +209,20 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
 
     for (std::vector<librbd::snap_info_t>::const_iterator snap =
          snap_list.begin(); snap != snap_list.end(); ++snap) {
-      librbd::Image snap_image;
+      librbd::Image snap_image;     
       r = rbd.open_read_only(io_ctx, snap_image, image_spec.name.c_str(),
-                             snap->name.c_str());
+                             nullptr);
       if (r < 0) {
         std::cerr << "rbd: error opening snapshot " << image_spec.name << "@"
```

Comment:
This error message needs adjustment, as we no longer open a snapshot here.  I'd change it to `rbd: error opening image` and dump the snapshot ID in the `rbd: error setting snapshot` error message below.